src/runtimes.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable no-var */
+declare global {
+  var EdgeRuntime: any;
+  var Netlify: any;
+  var Bun: any;
+  var Deno: any;
+  var __lagon__: any;
+  var fastly: any;
+}
+
+// https://runtime-keys.proposal.wintercg.org/
+export type Runtime =
+  | "workerd"
+  | "deno"
+  | "lagon"
+  | "netlify"
+  | "node"
+  | "bun"
+  | "edge-light"
+  | "fastly"
+  | "";
+
+// https://developers.cloudflare.com/workers/runtime-apis/web-standards/#navigatoruseragent
+const CLOUDFLARE_WORKERS_NAVIGATOR = "Cloudflare-Workers";
+
+// https://nodejs.org/api/process.html#processrelease
+const NODE_PROCESS_RELEASE_NAME = "node";
+
+export function detectRuntime(): Runtime {
+  if (typeof Netlify === "object") {
+    return "netlify";
+  }
+
+  if (typeof EdgeRuntime === "string") {
+    return "edge-light";
+  }
+
+  if (globalThis.navigator?.userAgent === CLOUDFLARE_WORKERS_NAVIGATOR) {
+    return "workerd";
+  }
+
+  if (globalThis.Deno) {
+    return "deno";
+  }
+
+  if (globalThis.__lagon__) {
+    return "lagon";
+  }
+
+  if (globalThis.process?.release?.name === NODE_PROCESS_RELEASE_NAME) {
+    return "node";
+  }
+
+  if (globalThis.Bun) {
+    return "bun";
+  }
+
+  if (globalThis.fastly) {
```

Comment:
Perf: We could "compress" runtime detections into an array by iterating over to reduce bundle size.

Reply:
If we want to be precise, checking against a global isn't the best way to detect which runtime is which. Some checks are already different (node & workerd) and ideally, we should avoid checking for the presence of global variables.

Reply:
Yes, what I mean is to take a similar approach of platforms for minimizing the checks that (currently) are against `globalThis` and merge them. For exceptions, we can always use different context. Let me know if you are happy i can push what i mean 👍🏼

Reply:
I've updated some checks to not rely on `globalThis`. Feel free to push what you're thinking about!

Reply:
Thank's! I have pushed a refactor including to use array for detectors.